Formula/b/bsc.rb

```diff
@@ -0,0 +1,109 @@
+class Bsc < Formula
+  desc "Bluespec Compiler (BSC)"
+  homepage "https://github.com/B-Lang-org/bsc"
+  url "https://github.com/B-Lang-org/bsc.git",
+    tag:      "2025.01.1",
+    revision: "65e3a87a17f6b9cf38cbb7b6ad7a4473f025c098"
+  license "BSD-3-Clause"
+  head "https://github.com/B-Lang-org/bsc.git", branch: "main"
+
+  depends_on "autoconf" => :build
+  depends_on "cabal-install" => :build
+  depends_on "ghc" => :build
+  depends_on "gperf" => :build
+  depends_on "make" => :build
+  depends_on "pkgconf" => :build
+  depends_on "gmp"
+  depends_on "icarus-verilog"
+  depends_on "tcl-tk@8"
+
+  uses_from_macos "bison" => :build
+  uses_from_macos "flex" => :build
+  uses_from_macos "libffi"
```

Comment:
You may wish to add a dependency on Perl as [the INSTALL.md file](https://github.com/B-Lang-org/bsc/blob/main/INSTALL.md) describes, for the vendored STP dependency, although if it's currently working then maybe it's fine?